### PR TITLE
AWS SNS/SQS PubSub Certification Tests - Dead Letter Queue

### PR DESCRIPTION
**Pull request by robertojrojas**
# Description
Provides the AWS SNS/SQS PubSub Certification Test Plan and Test Code. 
Adds Dead Letter Queue test.

## Issue reference
This PR makes progress towards #1339 and #1971 but does not complete these. More component metadata tests will need to be added.

### CODE REVIEW

**Comment by berndverst** [comment]
@artursouza `/ok-to-test` does not reflect workflow changes. There is no point to run that command.

Let's merge this PR after 1.10 is cut Friday. As we discussed before that will not impact our ability to certify this component as stable.

File: tests/certification/pubsub/aws/snssqs/snssqs_test.go [hunk 1]
```diff
@@ -1354,6 +1370,157 @@ func SNSSQSFIFOMessages(t *testing.T) {
...
+			return nil
+		}
+	}
+
+	assertMessages := func(timeout time.Duration, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// assert for messages
+			for _, m := range messageWatchers {
+				if !m.Assert(ctx, 3*timeout) {
+					ctx.Errorf("SNSSQSMessageDeadLetter - message assersion failed for watcher: %#v\n", m)
```
Comment (artursouza) [inline tests/certification/pubsub/aws/snssqs/snssqs_test.go:1482]: typo: assertion

File: tests/certification/pubsub/aws/snssqs/snssqs_test.go [hunk 2]
```diff
@@ -1354,6 +1370,157 @@ func SNSSQSFIFOMessages(t *testing.T) {
...
+		// Run the Dapr sidecar with ConsumerID "PUBSUB_AWS_SNSSQS_QUEUE_DLIN"
+		Step(sidecar.Run(subAppSideCar,
+			embedded.WithComponentsPath("./components/deadletter"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset+4),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset+4),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset+4),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset+4),
+			componentRuntimeOptions(),
+		)).
+		Step("publish messages to deadLetterTopicIn ==> "+deadLetterTopicIn, publishMessages(nil, subAppSideCar, deadLetterTopicIn, deadLetterConsumerGroup)).
```
Comment (artursouza) [inline tests/certification/pubsub/aws/snssqs/snssqs_test.go:1516]: Not sure If I am reading this right, why publishing directly to DLQ and not publishing to the regular topic and waiting for it to expire into DLQ?

**Comment by artursouza** [comment]
> @artursouza `/ok-to-test` does not reflect workflow changes. There is no point to run that command.
> 
> Let's merge this PR after 1.10 is cut Friday. As we discussed before that will not impact our ability to certify this component as stable.

Saw that now. Can we move those long commands into scripts? This way, we can at least move these changes outside workflows going forward.

**Comment by berndverst** [comment]
Yes we can but let's do that after the 1.10 release.

File: tests/certification/pubsub/aws/snssqs/snssqs_test.go (hunk 2 above)
Comment (robertojrojas) [reply tests/certification/pubsub/aws/snssqs/snssqs_test.go:1516]: Actually, it's publishing to a Topic. There is a typo in a comment above that line and it should read: `PUBSUB_AWS_SNSSQS_TOPIC_DLIN`. That it's an environment variable that will populate the variable `deadLetterTopicIn` at the time the workflow runs.

**Review by berndverst — APPROVED** [review]

**Pull request merged**